### PR TITLE
fix(hr): Payroll Entry with payable account types

### PR DESCRIPTION
**Pull request by ghost**
Currently, the Payroll Entry doctype fails to create an accrual Journal Entry if any of the relevant accounts are of type Payable. 

See here for relevant discussion: https://discuss.erpnext.com/t/v11-payroll-entry-party-type-party-required-for-payable-account-salaries-payable/44931

This pull request modifies the logic of Payroll Entry docs to break Journal Entry lines down by employee when (and only when) an accrual account is marked Payable.

### CODE REVIEW

**Comment by theopen-institute** [comment]
Hello, anything I can do to help facilitate this pull request?

**Comment by theopen-institute** [comment]
Updated to rebase following PR #21609

**Comment by theopen-institute** [comment]
Hi @Anurag810. This PR has been open for almost three weeks with no response. Is there anything I can do to help get it reviewed?

**Review by nabinhait — CHANGES_REQUESTED** [review]
Currently, we do consolidated accrual to hide salary data from Accounts users.

I think we should add a setting “Enable Salary Accrual Per Employee” in Payroll Settings.
And based on that we should decide per employee accrual or consolidated accrual.
If it is per employee accrual, account_type should be forced as payable, otherwise not.

**Review by deepeshgarg007 — COMMENTED** [review]
Also we have moved the Payroll Entry doctype to separate payroll module. Please also consider that while making the changes

**Comment by theopen-institute** [comment]
> Currently, we do consolidated accrual to hide salary data from Accounts users.

Indeed, and that's important for many companies. Many other companies, however, do payroll directly in their main book of accounts, and the current system is quite awkward for them.

> I think we should add a setting “Enable Salary Accrual Per Employee” in Payroll Settings.
> And based on that we should decide per employee accrual or consolidated accrual.
> If it is per employee accrual, account_type should be forced as payable, otherwise not.

I think that's fine. I'd propose the following:

- Enable Salary Accrual Per Employee == false:
*Current behavior - All accruals posted to Journal Entry as aggregate*

- Enable Salary Accrual Per Employee == true:
*Payable accounts will be accrued per employee, and non-payable accounts will be accrued in aggregate. (For many use cases, it's important that some payroll expenses be per employee while others are not.)

If that sounds alright, I can go ahead and make the changes.

**Comment by theopen-institute** [comment]
> Also we have moved the Payroll Entry doctype to separate payroll module. Please also consider that while making the changes

Yes, I've seen that. From what I can tell, the only thing that will be affected is reference naming. Is there a timeframe for merging #21990?

**Comment by theopen-institute** [comment]
Hello, the requested changes have been made. If there are any next steps to getting this merged, please advise.

**Comment by theopen-institute** [comment]
@nabinhait Anything I can do to help get this merged?

**Comment by theopen-institute** [comment]
This silence is frustrating.

**Comment by Anurag810** [comment]
@theopen-institute sorry for the inconvenience. 
Actually, We are separating the payroll from Hr. we will merge this after, merging Payroll module.

File: erpnext/hr/doctype/hr_settings/hr_settings.json
Comment (Anurag810) [inline erpnext/hr/doctype/hr_settings/hr_settings.json:25]: The payroll module is merged please rebase your PR and push again.
and move the setting to new Payroll settings.

File: erpnext/hr/doctype/payroll_entry/payroll_entry.js
Comment (Anurag810) [inline erpnext/hr/doctype/payroll_entry/payroll_entry.js:235]: Use ```frm``` instead of ```cur_frm```

File: erpnext/hr/doctype/payroll_entry/payroll_entry.py
Comment (Anurag810) [inline erpnext/hr/doctype/payroll_entry/payroll_entry.py:242]: I think You can comonify the code for earnings and deduction by passing ```args ``` it is quite repetitive.

File: erpnext/hr/doctype/payroll_entry/payroll_entry.py
Comment (Anurag810) [inline erpnext/hr/doctype/payroll_entry/payroll_entry.py:275]: I think You can comonify the code for earnings and deduction by passing ```args ``` it is quite repetitive.

**Review by Anurag810 — CHANGES_REQUESTED** [review]

**Comment by Anurag810** [comment]
Please Reopen This PR When It is completed.

File: erpnext/hr/doctype/payroll_entry/payroll_entry.js
Comment (theopen-institute) [reply erpnext/hr/doctype/payroll_entry/payroll_entry.js:235]: Done

File: erpnext/hr/doctype/hr_settings/hr_settings.json
Comment (theopen-institute) [reply erpnext/hr/doctype/hr_settings/hr_settings.json:25]: Completed

File: erpnext/hr/doctype/payroll_entry/payroll_entry.py
Comment (theopen-institute) [reply erpnext/hr/doctype/payroll_entry/payroll_entry.py:242]: It is a bit repetitive, but the code in each instance is different enough that factoring out into a method with arguments gets messy. I'm happy to give it another shot if you have a specific implementation in mind, but I think the code is actually cleaner this way.

File: erpnext/hr/doctype/payroll_entry/payroll_entry.py
Comment (theopen-institute) [reply erpnext/hr/doctype/payroll_entry/payroll_entry.py:275]: See above.

**Comment by theopen-institute** [comment]
I've shifted the code over to the new Payroll module @Anurag810 Would you be able to reopen this PR?

**Comment by theopen-institute** [comment]
Hello, @Anurag810 is it possible to reopen this PR?

**Comment by theopen-institute** [comment]
Is there any reason this PR should not be reopened. If these are not changes you want, that's fine, but please just tell me. I know this isn't a huge contribution, but I am new to frappe and it actually took me quite a bit of time to figure out. The fact that I don't even get a response is very disheartening.

@Anurag810 @nabinhait @rohitwaghchaure @deepeshgarg007

**Comment by theopen-institute** [comment]
This is so frustrating...

@Anurag810 @nabinhait @rohitwaghchaure @deepeshgarg007

**Comment by mhbu50** [comment]
I need this also, It is very important for us
Any help?
@Anurag810 @nabinhait @rohitwaghchaure @deepeshgarg007

**Comment by nabinhait** [comment]
Sorry, the PR cannot be reopened from our side.

Also, the code is a little bit repetitive, we will take this project next month and it will be part of v13.

**Comment by theopen-institute** [comment]
The permissions on your repository are the issue, because I can't reopen it either.

Regarding the code, I asked several times for guidance on the approach you wanted me to improve it, but I got no response for weeks at a time. If you'd rather just do it yourself, that's fine.

**Pull request closed without merging**
